lib/Transforms/Instrumentation/ShadowMemSeaDsa.cc

```diff
@@ -1357,11 +1396,17 @@ void ShadowDsaImpl::solveUses(Function &F) {
 
 namespace seahorn {
 
+ShadowMemSeaDsa::ShadowMemSeaDsa() : llvm::ModulePass(ID), m_shadow(nullptr) {
+ m_dsa = NULL;
+}
+
+ShadowMemSeaDsa::~ShadowMemSeaDsa() {}
+
 bool ShadowMemSeaDsa::runOnModule(llvm::Module &M) {
   if (M.begin() == M.end())
     return false;
 
-  dsa::GlobalAnalysis &dsa = getAnalysis<dsa::DsaAnalysis>().getDsaAnalysis();
+  m_dsa = &getAnalysis<dsa::DsaAnalysis>().getDsaAnalysis();
```

Comment:
use `getAnalysisIfAvailable<>` which only returns analysis if it has been scheduled by the pass manager.
On the other hand, `getAnalysis<>` will schedule an additional run of the analysis if necessary